golang/go-cloud-run-hello-world/main.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+// templateData provides template parameters.
+type templateData struct {
+	Service      string
+	Revision     string
+	Project      string
+	ProjectFound bool
+}
+
+// Variables used to generate the HTML page.
+var (
+	data templateData
+	tmpl *template.Template
+)
+
+func main() {
+	// Comment out in local development for timestamp prefixes.
```

Comment:
Can this be implemented such that there will be no required manual step to get this to run locally vs on Cloud Run?

Reply:
This isn't required to run locally. It's just nice to have in terminal output, whereas it's a distraction in stackdriver.

Options:
1. Remove and always have timestamps
2. Leave in place and let developers edit the code if they want to.
3. Add a check for something like `DEV=1` env var and toggle accordingly.

The plan is for Cloud Code to simulate all the mechanisms we might use to detect running in Cloud Run.

Reply:
Could we generalize option 3 a little bit? So instread of introducing a new environment variable for local vs. cloud, we can test the same logic you already have for determining whether to call metadata server.

Reply:
Leaning on GOOGLE_CLOUD_PROJECT env var for this. It's a bit awkward for conceptual flow but adaptive is better than commenting instructions.